execution-engine/contracts/hdac-system/pop/src/pop_impl/mod.rs

```diff
@@ -274,8 +274,29 @@ impl ProofOfProfessionContract {
         // In the fare system, the fee is taken by the validator.
         let reward_purse =
             get_purse(uref_names::POS_REWARD_PURSE).map_err(PurseLookupError::rewards)?;
+        let commission_purse =
+            get_purse(uref_names::POS_COMMISSION_PURSE).map_err(PurseLookupError::commission)?;
 
-        system::transfer_from_purse_to_purse(payment_purse, reward_purse, total)
+        let reward_amount =
+            total * sys_params::VALIDATOR_COMMISSION_RATE_IN_PERCENTAGE / U512::from(100);
+        let commission_amount = total
+            * (U512::from(100) - sys_params::VALIDATOR_COMMISSION_RATE_IN_PERCENTAGE)
+            / U512::from(100);
+
+        if total != (reward_amount + commission_amount) {
+            let remain_amount = total - reward_amount - commission_amount;
+
+            let communtiy_purse =
+                get_purse(uref_names::POS_COMMUNITY_PURSE).map_err(PurseLookupError::communtiy)?;
+
+            system::transfer_from_purse_to_purse(payment_purse, communtiy_purse, remain_amount)
+                .map_err(|_| Error::FailedTransferToRewardsPurse)?;
```

Comment:
It seems like a wrong error, `Error::FailedTransferToRewardsPurse` -> `Error::FailedTransferToCommunityPurse`

Reply:
https://github.com/hdac-io/CasperLabs/pull/30/commits/653d226b40ca1529d6d067b382f03e42199c967d